addons/account/models/sequence_mixin.py

```diff
@@ -25,6 +25,7 @@ class SequenceMixin(models.AbstractModel):
     _sequence_index = False
     _sequence_monthly_regex = r'^(?P<prefix1>.*?)(?P<year>((?<=\D)|(?<=^))((19|20|21)\d{2}|(\d{2}(?=\D))))(?P<prefix2>\D*?)(?P<month>(0[1-9]|1[0-2]))(?P<prefix3>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
     _sequence_yearly_regex = r'^(?P<prefix1>.*?)(?P<year>((?<=\D)|(?<=^))((19|20|21)?\d{2}))(?P<prefix2>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
+    _sequence_yearly_financial_regex = r'^(?P<prefix1>.*?)(?P<fyear_start>((?<=\D)|(?<=^))((19|20|21)?\d{2}))(?P<prefix2>([^0-9a-zA-Z\/]{1}))(?P<fyear_end>((?<=\D)|(?<=^))((19|20|21)?\d{2}))(?P<prefix3>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
```

Comment:
This works for years between `2010` and `2099` but how can you differentiate between a reset of `month` or `fyear_start` for a sequence like `INV/10-11/00001`?
Is it the fiscal year 2010-2011 or the month of November in 2010?

The only solution I see is to use the full year instead of only the last 2 digits (also simplifying multiple parts of the code)

Reply:
Yes, this is a conflict,
But practically right now on one entry of 2010 bills and until 2099 It's improved a lot :smile: 
a full year is not practical because in India the length of an Invoice number is 16
`INV/2022-2023/99` only 99 invoices

Reply:
then how to use this type of sequence?

Reply:
We can't and we never will be. At least not with 2 digit years.

Reply:
:thinking: hmm, let's go with full year.
if customers need 2 digit years then we handle it with custom regexes :wink:

Reply:
That works for me then.
At least it will be explicit for what we are expecting

Reply:
@william-andre code pushed plz check